src/drawer.js

```diff
@@ -45,21 +45,22 @@ WaveSurfer.Drawer = {
         var clientX = e.targetTouches ? e.targetTouches[0].clientX : e.clientX;
         var bbox = this.wrapper.getBoundingClientRect();
 
-        var nominalWidth = this.width;
-        var parentWidth = this.getWidth();
+        var visibleWidth = this.width;
+        var scrollWidth = this.getWidth();
 
-        var progress;
-
-        if (!this.params.fillParent && nominalWidth < parentWidth) {
-            progress = ((clientX - bbox.left) * this.params.pixelRatio / nominalWidth) || 0;
-
-            if (progress > 1) {
-                progress = 1;
-            }
+        // If the entire container is not filled and further if the nominal width is less than the parent width...
+        if (!this.params.fillParent && visibleWidth < scrollWidth) {
+            var numerator = (clientX - bbox.left) * this.params.pixelRatio; // Apparently we need to scale this here...
+            var denominator = visibleWidth - 1;
         } else {
-            progress = ((clientX - bbox.left + this.wrapper.scrollLeft) / this.wrapper.scrollWidth) || 0;
+            var numerator = clientX - bbox.left + this.wrapper.scrollLeft;
```

Comment:
No, because `visibleWidth` is the actual size and (just renamed) `scrollContainerWidth` is a virtual size which doesn't change if the pixelRatio goes up or down. I am not sure about the if statement though.

In general, params.fillParent is a bit difficult to understand.